### PR TITLE
Reconfigure surefire plugin

### PR DESCRIPTION
**Pull request by gscheibel**
to be sure that all the unit tests are executed properly

### CODE REVIEW

File: src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt [hunk 1]
```diff
@@ -66,8 +61,13 @@ internal fun KAnnotatedElement.isGraphQLIgnored() = this.findAnnotation<GraphQLI
 
 internal fun KAnnotatedElement.isGraphQLID() = this.findAnnotation<GraphQLID>() != null
 
-internal fun Annotation.getDirectiveInfo(): DirectiveAnnotation? =
-    this.annotationClass.annotations.find { it is DirectiveAnnotation } as? DirectiveAnnotation
+private fun Annotation.getDirectiveInfo(): DirectiveInfo? {
+    val directiveAnnotation = this.annotationClass.annotations.find { it is DirectiveAnnotation } as? DirectiveAnnotation
```
Comment (d4rken) [inline src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt]: What's the benefit of these changes (this and the following lines)? Why are they problematic for the surefire plugin?

Comment (dkuc84) [reply src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt]: indeed this is unrelated to surefire plugin configuration. It is addressing another bug

Comment (d4rken) [reply src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt]: What was the bug? I changed the same code in #69 and need to resolve the merge conflict.

Comment (gscheibel) [reply src/main/kotlin/com/expedia/graphql/schema/extensions/annotationExtensions.kt]: The problem here was when we have to set the name of the directive there are 2 options:
- from the `name` property of the `GraphQLDirective` annotation
- the name of the custom annotation (which is annotated with `@GraphQLDirective` 

However as we extracted the `@GraphQLDirective` type we lost access to the name of the custom directive class name.
So I extracted it into a new class and centralized the logic in the `effectiveName` method as it's shared with the description logic

**Pull request merged**